### PR TITLE
Remove and refactor legacy receipt code 

### PR DESCRIPTION
**Pull request by pathornteng**
This PR refactors receipt code to make it cleaner. 

Ref: https://github.com/loomnetwork/loomchain/issues/1240

- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

### CODE REVIEW

File: receipts/handler/handler.go [hunk 1]
```diff
@@ -18,86 +17,45 @@ import (
...
-			return nil, errors.Wrap(err, "new leved db receipt handler")
-		}
-		rh.leveldbReceipts = leveldbHandler
-	}
-	return rh, nil
 }
 
 func (r *ReceiptHandler) Version() ReceiptHandlerVersion {
-	return r.v
+	return ReceiptHandlerLevelDb
```
Comment (enlight) [inline receipts/handler/handler.go]: Is this used anywhere?

Comment (pathornteng) [reply receipts/handler/handler.go]: Umm, look like it is an unused function, removed.

File: receipts/handler/handler.go [hunk 2]
```diff
@@ -18,86 +17,41 @@ import (
...
-			receipt, chainErr = r.chainReceipts.GetReceipt(state, txHash)
-			if chainErr != nil {
-				return receipt, errors.Wrapf(common.ErrTxReceiptNotFound, "%v: %v", chainErr, err)
-			}
-		}
-		return receipt, nil
+func (r *ReceiptHandler) GetReceipt(txHash []byte) (types.EvmTxReceipt, error) {
+	receipt, err := r.leveldbReceipts.GetReceipt(txHash)
+	if err != nil {
+		return receipt, errors.Wrapf(common.ErrTxReceiptNotFound, "%v: %v", err, err)
```
Comment (enlight) [inline receipts/handler/handler.go]: Huh? What's with the `err, err`?

File: receipts/handler/handler.go [hunk 3]
```diff
@@ -120,113 +74,71 @@ func (r *ReceiptHandler) GetCurrentReceipt() *types.EvmTxReceipt {
...
-		r.mutex.RUnlock()
-	default:
-		err = loomchain.ErrInvalidVersion
-	}
-
 	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	if err := r.leveldbReceipts.CommitBlock(state, r.receiptsCache, uint64(height)); err != nil {
+		return err
+	}
```
Comment (enlight) [inline receipts/handler/handler.go]: This is different to how it worked before, before the hash list & receipts cache were reset regardless of whether or not `err == nil`. You think the old behavior was wrong?

Comment (pathornteng) [reply receipts/handler/handler.go]: No, the behavior should be the same. Fixed

File: receipts/handler/handler.go (hunk 2 above)
Comment (pathornteng) [reply receipts/handler/handler.go]: Fixed

**Pull request merged**